File: Settings/src/org/nameless/settings/system/ManageGamesFragment.java
/*
 * Copyright (C) 2024 The Nameless-AOSP Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.nameless.settings.system;

import android.os.Bundle;

import com.android.settings.R;

import java.util.List;

import org.nameless.app.GameModeManager;
import org.nameless.custom.preference.SwitchPreference;
import org.nameless.settings.fragment.PerAppSwitchConfigFragment;

public class ManageGamesFragment extends PerAppSwitchConfigFragment {

    private GameModeManager mGameModeManager;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mGameModeManager = getActivity().getSystemService(GameModeManager.class);
    }

    @Override
    protected int getAllowedSystemAppListResId() {
        return 0;
    }

    @Override
    protected int getPreferenceScreenResId() {
        return R.xml.manage_games;
    }

    @Override
    protected int getTopInfoResId() {
        return R.string.game_mode_manage_apps_summary;
    }

    @Override
    protected boolean isChecked(String packageName, int uid) {
        return mGameModeManager.isAppGame(packageName);
    }

    @Override
    protected boolean onSetChecked(SwitchPreference pref, String packageName, int uid, boolean checked) {
        if (checked) {
            mGameModeManager.addGame(packageName);
        } else {
            mGameModeManager.removeGame(packageName);
        }
        return true;
    }

    @Override
    protected void onCheckedListUpdated(List<String> pkgList) {}
}

File: Settings/src/org/sun/settings/system/ManageGamesFragment.kt
/*
 * Copyright (C) 2025 The Nameless-CLO Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.sun.settings.system

import com.android.settings.R

import org.sun.app.GameModeManager
import org.sun.custom.preference.SwitchPreferenceCompat
import org.sun.settings.fragment.PerAppSwitchConfigFragment

class ManageGamesFragment : PerAppSwitchConfigFragment() {

    private val gameModeManager by lazy { requireContext().getSystemService(GameModeManager::class.java)!! }

    override fun getAllowedSystemAppListResId() = 0

    override fun getTitleResId() = R.string.game_mode_manage_apps_title

    override fun getTopInfoResId() = R.string.game_mode_manage_apps_summary

    override fun isChecked(packageName: String, uid: Int): Boolean {
        return gameModeManager.isAppGame(packageName)
    }

    override fun onSetChecked(pref: SwitchPreferenceCompat, packageName: String, uid: Int, checked: Boolean): Boolean {
        if (checked) {
            gameModeManager.addGame(packageName)
        } else {
            gameModeManager.removeGame(packageName)
        }
        return true;
    }

    override fun onCheckedListUpdated(pkgList: List<String>) {}

    override fun getSelectAllRunnable(): Runnable? {
        return Runnable {
            super.getSelectAllRunnable()?.run()
            allAppData.forEach {
                gameModeManager.addGame(it.packageName)
            }
        }
    }

    override fun getDeselectAllRunnable(): Runnable? {
        return Runnable {
            super.getDeselectAllRunnable()?.run()
            allAppData.forEach {
                gameModeManager.removeGame(it.packageName)
            }
        }
    }
}
